### DISCUSSION TITLE
V3 - How to deactivate multi-select functionality only when a specific option is chosen?

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#9680] [Help] [answered]
I want to conditionally apply the `->multiple()` method based on the selected value of the `colors` field. If the selected value is `others`, I want to remove the `->multiple()` method.

```
protected function getColorsFormComponent(): Component
    {
        return Group::make()
            ->schema([
                Select::make('colors')
                    ->label(__('What is your favorite color?'))
                    ->options([
                        'blue' => __('Blue'),
                        'red' => __('Red'),
                        'pink' => __('Pink'),
                        'brown' => __('Brown'),
                        'others' => __('Others'),
                    ])
                    ->native(false)
                    ->searchable()
                    ->minItems(1)
                    ->maxItems(5)
                    ->required()
                    ->live()
                    ->multiple(function (Get $get) {
                        return $get('colors') !== 'others';
                    })
                }),
                TagsInput::make('other_colors')
                    ->label('')
                    ->placeholder(__('Enter your other colors'))
                    ->hidden(fn (Get $get) => $get('colors') !== 'others')
                    ->required(fn (Get $get) => $get('colors') === 'others')
                    ->notIn('colors')
                    ->nestedRecursiveRules([
                        'min:1',
                        'max:5',
                    ])
                    ->validationAttribute(__('other colors')),
            ]);
    }
```

**Comment by sinnbeck** [comment]
Why not instead have a checkbox to toggle showing/hiding the select

- [ ] I want to pick a color

**Reply by devhoussam1998** [reply]
is there any way to achieve that by my example above?

**Reply by sinnbeck** [reply]
What I mean is something like this: https://filamentphp.com/docs/3.x/forms/advanced#conditionally-hiding-a-field

**Reply by devhoussam1998** [reply]
did you understand what I want to achieve?

_I want to conditionally apply the ->multiple() method based on the selected value of the colors field. If the selected value is others, I want to remove the ->multiple() method._

**Reply by sinnbeck** [reply]
Yeah I get that. But as that will require you to reimplement the select field as it isn't possible to reinitialize choices-js after the first render. So I gave a suggestion to another way of doing the logic

**Reply by devhoussam1998** [reply]
@sinnbeck based on what you said what I want to achieve is not possible?
well thank you for your advice I will try to impliment that

**Reply by sinnbeck** [reply, marked as answer]
Correct. One idea I just had.. Have 2 selects and sync up what is selected in them. One is ->multiple() and one is not. Then when you select `Others`, you hide the mult iselect one and show the other and the other way around

**Reply by devhoussam1998** [reply]
thank you so mush @sinnbeck
